src/ship/controller/physicaldevice/ConnectedPhysicalDeviceManager.cpp

```diff
@@ -65,9 +75,31 @@ void ConnectedPhysicalDeviceManager::RefreshConnectedSDLGamepads() {
         mConnectedSDLGamepads[instanceId] = gamepad;
         mConnectedSDLGamepadNames[instanceId] = name;
 
-        for (uint8_t port = 1; port < 4; port++) {
-            mIgnoredInstanceIds[port].insert(instanceId);
+#ifdef ENABLE_EXP_AUTO_CONFIGURE_CONTROLLERS
+        // Check if auto-configure controllers is enabled (default: enabled/1)
+        bool autoConfigureEnabled =
+            Context::GetInstance()->GetConsoleVariables()->GetInteger(CVAR_AUTO_CONFIGURE_CONTROLLERS, 0);
+
+        // Only auto-assign controllers to specific ports if enabled
+        if (autoConfigureEnabled) {
+            for (uint8_t j = 0; j < 4; j++) {
+                if (port == j) {
+                    continue;
+                }
+                mIgnoredInstanceIds[j].insert(instanceId);
+            }
+            port++;
+        } else {
+            for (uint8_t p = 1; p < 4; p++) {
+                mIgnoredInstanceIds[p].insert(instanceId);
+            }
+        }
+#else
+        for (uint8_t j = 1; j < 4; j++) {
+            mIgnoredInstanceIds[j].insert(instanceId);
```

Comment:
with the above suggestion we can keep this as it was

```suggestion
        for (uint8_t port = 1; port < 4; port++) {
            mIgnoredInstanceIds[port].insert(instanceId);
```

Reply:
same as above